package-builder/build-manifest.nix

```diff
@@ -0,0 +1,149 @@
+          signDarwinBinariesInAllOutputs
+        ''}
+      ''
+      else ''
+        echo "---"
+        echo "Input checksums:"
+        md5sum \
+          ${/. + srcTarball} \
+          ${buildScript-contents} \
+          ${pkgs.lib.optionalString (
+            buildCache-tar-contents != null
+          )
+          buildCache-tar-contents}
+        echo "---"
+        # If the build script is provided, then it's expected that we will
+        # invoke it from within the sandbox to create $out. The choice of
+        # pure or impure mode occurs outside of this script as the derivation
+        # is instantiated.
+        source $stdenv/setup # is this necessary?
+
+        # We are currently in /build, and TMPDIR is also set to /build, so
+        # we need to extract the source and work in a subdirectory to avoid
+        # populating our build cache with a bunch of temporary files.
+        mkdir $name && cd $name
+
+        # We pass and extract the source as a tarball to preserve timestamps.
+        # Passing the source as a directory would cause the timestamps to be
+        # set to the UNIX epoch as happens with all files in the Nix store,
+        # which would be older than the intermediate compilation artifacts.
+        tar -xpf ${/. + srcTarball}
+
+        # Extract contents of the cache, if it exists.
+        ${
+          if buildCache-tar-contents == null
+          then ":"
+          else ''
+            tar --skip-old-files -xpf ${buildCache-tar-contents}
+          ''
+        }
+        ${
+          if buildCache == null
+          then ''
+                 # When not preserving a cache we just run the build normally.
+                 FLOX_SRC_DIR=$(pwd) ${flox-env-package}/activate --turbo -- \
+            bash -e ${buildScript-contents}
+          ''
+          else ''
+                 # If the build fails we still want to preserve the build cache, so we
+                 # remove $out on failure and allow the Nix build to proceed to write
+                 # the result symlink.
+                 FLOX_SRC_DIR=$(pwd) ${flox-env-package}/activate --turbo -- \
+            bash -e ${buildScript-contents} || \
+                   ( rm -rf $out && echo "flox build failed (caching build dir)" | tee $out 1>&2 )
+          ''
+        }
+      ''
+    )
+    + ''
+      # Start by patching shebangs in bin and sbin directories.
+      for dir in $out/bin $out/sbin; do
+        if [ -d "$dir" ]; then
+          patchShebangs $dir
+        fi
+      done
+      # Wrap contents of files in bin with ${flox-env-package}/activate
+      for prog in $out/bin/* $out/sbin/*; do
+        if [ -L "$prog" ]; then
+          : # You cannot wrap a symlink, so just leave it be?
+        else
+          assertExecutable "$prog"
+          hidden="$(dirname "$prog")/.$(basename "$prog")"-wrapped
+          mv "$prog" "$hidden"
+          makeShellWrapper "${flox-env-package}/activate" "$prog" \
```

Comment:
note nonblocking: I'm expecting the wrapper should only include the packages in the `toplevel` package group, but not vars, etc-profiles, or hooks. Is that right?

Reply:
Correct .. once the env builder is able to create a [deep recursive] buildenv of only the toplevel package group, _and_ it can provide the path to that output as part of the build invocation then yes we will use that [smaller closure] package to wrap the binaries.  👍 

Or to be slightly more specific, `flox-env-package` in the above line will eventually be replaced with `flox-env-toplevel-package` (or `flox-env-runtime-package`?) once our flox env builder has been updated to do that.